networkit/cpp/io/NetworkitBinaryWriter.cpp

```diff
@@ -17,12 +18,23 @@
 
 namespace NetworKit {
 
-NetworkitBinaryWriter::NetworkitBinaryWriter(uint64_t chunks, NetworkitBinaryWeights weightsType): chunks(chunks), weightsType(weightsType) {}
+NetworkitBinaryWriter::NetworkitBinaryWriter(uint64_t chunks, NetworkitBinaryWeights weightsType, bool preserveEdgeIndex): chunks(chunks), weightsType(weightsType), preserveEdgeIndex(preserveEdgeIndex) {}
 
 void NetworkitBinaryWriter::write(const Graph &G, const std::string &path) {
-
     std::ofstream outfile(path, std::ios::binary);
     Aux::enforceOpened(outfile);
+    writeData(outfile, G);
+    INFO("Written graph to ", path);
+}
+
+std::string NetworkitBinaryWriter::writeState(const Graph &G) {
+    std::stringstream outdata;
+    writeData(outdata, G);
+    return outdata.str();
+}
+
+template <class T>
+void NetworkitBinaryWriter::writeData(T &outStream, const Graph &G){
```

Comment:
In this case, the MemoryMappedFile functionality of Networkit would need to be expanded as it currently can only put existing files into memory (and will fail when opening empty files) and has currently no write functionality.

Reply:
yes, it works since string to std::vector<uint8_t> is a easy conversion. So no problem on the c++ side. I currently just need to test with cython since the documentation recommends using std::string when interfacing bytes data with python.